agent/bench-scripts/pbench-linpack

```diff
@@ -138,16 +166,36 @@ fi
 
 trap "interrupt" INT QUIT TERM
 
-## Run the benchmark and start/stop perf analysis tools
-pbench-start-tools --group=$tool_group --dir=$benchmark_results_dir
-
-OMP_NUM_THREADS=$threads ${linpack_dir}/${linpack_cmd} < ${linpack_dir}/${linpack_dat} | tee $result_file
-
-pbench-stop-tools --group=$tool_group --dir=$benchmark_results_dir
-pbench-send-tools --group=$tool_group --dir=$benchmark_results_dir
-pbench-postprocess-tools --group=$tool_group --dir=$benchmark_results_dir
-
-ln -s ${sample} $benchmark_run_dir/$iteration/reference-result
+let count=1
+for thread in ${threads//,/ }; do
+	iteration="$count-$thread-threads"
+	echo $iteration >> $benchmark_iterations
+	echo "Starting iteration ${iteration}"
+	# Generate the linpack.input dirs (in /var/lib/pbench-agent/tmp)
+	iteration_dir="$benchmark_run_dir/$iteration"
+	mkdir -p "$iteration_dir"
+	store_and_run "$iteration_dir" "'$pbench_bin/bench-scripts/postprocess/linpack-prepare-input-file' --output-dir '$iteration_dir' --threads '$thread' --run-samples 1 --linpack-binary '$linpack_dir/$linpack_cmd'" "$benchmark-preprocess.cmd"
+	for sample in $(seq 1 $nr_samples); do
+		echo "test sample $sample of $nr_samples"
+		sample_dir="$iteration_dir/sample$sample"
+		# Create the output directory expected by the postprocess script
+		mkdir -p "$sample_dir/clients/localhost/"
+
+		## Run the benchmark and start/stop perf analysis tools
+		pbench-start-tools --group=$tool_group --dir="$sample_dir"
+
+		( cd "$iteration_dir" && ./linpack.sh )
+		mv "$iteration_dir/"linpack.{out,meta} "$sample_dir/clients/localhost"
+
+		pbench-stop-tools --group=$tool_group --dir="$sample_dir"
+		pbench-send-tools --group=$tool_group --dir="$sample_dir"
+		pbench-postprocess-tools --group=$tool_group --dir="$sample_dir"
+		postprocess-iteration "$sample_dir" "$benchmark_run_dir"
```

Comment:
It's a small detail, but shouldn't we be stopping the tools before moving the output files?  (I realize that moving the files is probably a cheap operation, but we don't want that operation included in the data which the tools are collecting, right?)

Reply:
Yes, you are right, that's why reviews are so important :-)